internal/retry/retryer_test.go

```diff
@@ -284,3 +284,31 @@ func (suite *UnitTestSuite) TestRetryerWithEmptyCollectionName() {
 	suite.NoError(err)
 	suite.Equal("", name)
 }
+
+// Test fix for REP-4197
+func (suite *UnitTestSuite) TestRetryerWithUUIDNotSupportedError() {
+	retryer := New(0)
+
+	attemptNumber := 0
+	cmdErr := mongo.CommandError{
+		Message: "(Location4928902) collectionUUID is not supported on a mongos",
+		Name:    "FailedToParseError",
+		Code:    9,
+	}
```

Comment:
Is it possible to test against both v5 and pre-v5 variants of the error?

Reply:
Make sense! I added a test for pre-v5.